packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -413,23 +399,31 @@ export function injectIntoGlobalHook(globalObject: any): void {
     // Here, we just want to get a reference to scheduleRefresh.
     const oldInject = hook.inject;
     hook.inject = function(injected) {
-      findHostInstancesForRefresh = ((injected: any)
-        .findHostInstancesForRefresh: FindHostInstancesForRefresh);
-      scheduleRefresh = ((injected: any).scheduleRefresh: ScheduleRefresh);
-      scheduleRoot = ((injected: any).scheduleRoot: ScheduleRoot);
-      setRefreshHandler = ((injected: any)
-        .setRefreshHandler: SetRefreshHandler);
-      return oldInject.apply(this, arguments);
+      const id = oldInject.apply(this, arguments);
+      if (
+        typeof injected.scheduleRefresh === 'function' &&
+        typeof injected.setRefreshHandler === 'function'
+      ) {
+        // This version supports React Refresh.
+        helpersByRendererID.set(id, ((injected: any): RendererHelpers));
+      }
+      return id;
     };
 
     // We also want to track currently mounted roots.
     const oldOnCommitFiberRoot = hook.onCommitFiberRoot;
     hook.onCommitFiberRoot = function(
-      id: mixed,
+      id: number,
       root: FiberRoot,
       maybePriorityLevel: mixed,
       didError: boolean,
     ) {
+      const helpers = helpersByRendererID.get(id);
+      if (helpers === undefined) {
+        return;
```

Comment:
There can still be commits to renderers that don't support Fast Refresh. We ignore those.

Reply:
This might be nice to leave as a comment.